Build failure on Apple Silicon

the build error is:

```
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/t/util/test_timer_enabled.cpp:14:5: error: invalid token in expression
    REQUIRE(timer.elapsed_microseconds() > 900);
    ^
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/catch/catch.hpp:11594:25: note: expanded from macro 'REQUIRE'
#define REQUIRE( expr ) INTERNAL_CATCH_TEST( "REQUIRE", Catch::ResultDisposition::Normal, expr  )
                        ^
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/catch/catch.hpp:2235:9: note: expanded from macro 'INTERNAL_CATCH_TEST'
        INTERNAL_CATCH_REACT( __catchResult ) \
        ^
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/catch/catch.hpp:2219:44: note: expanded from macro 'INTERNAL_CATCH_REACT'
    if( resultBuilder.shouldDebugBreak() ) CATCH_BREAK_INTO_DEBUGGER(); \
                                           ^
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/catch/catch.hpp:2155:75: note: expanded from macro 'CATCH_BREAK_INTO_DEBUGGER'
    #define CATCH_BREAK_INTO_DEBUGGER() if( Catch::isDebuggerActive() ) { CATCH_TRAP(); }
                                                                          ^
/tmp/libosmium-20210111-77861-26tcd/libosmium-2.16.0/test/catch/catch.hpp:2133:38: note: expanded from macro 'CATCH_TRAP'
        #define CATCH_TRAP() __asm__("int $3\n" : : /* NOLINT */ )
                                     ^
<inline asm>:1:6: note: instantiated into assembly here
        int $3
            ^
```

Full logs at https://github.com/Homebrew/homebrew-core/pull/68774

----

The code in catch.hpp at https://github.com/osmcode/libosmium/blob/b293e96dbe2e3a9afe400768a990e107f8af3f02/test/catch/catch.hpp#L2128

does not have a valid codepath for `__aarch64__`, as provided in the latest catch version: see this commit https://github.com/catchorg/Catch2/commit/a25c1a24af8bffd35727a888a307ff0280cf9387

--- a/test/catch/catch.hpp
+++ b/test/catch/catch.hpp
@@ -2129,6 +2129,9 @@ namespace Catch{
         #define CATCH_TRAP() \
                 __asm__("li r0, 20\nsc\nnop\nli r0, 37\nli r4, 2\nsc\nnop\n" \
                 : : : "memory","r0","r3","r4" ) /* NOLINT */
+    #elif defined(__aarch64__)
+        // Based on https://github.com/catchorg/Catch2/commit/a25c1a24af8bffd35727a888a307ff0280cf9387
+        #define CATCH_TRAP() __asm__(".inst 0xd4200000")
     #else
         #define CATCH_TRAP() __asm__("int $3\n" : : /* NOLINT */ )
     #endif

